### ISSUE TITLE
Spec files are getting their own chunks

### ISSUE THREAD

**Issue by m-sawicki** [angular/angular-cli#10107]
```
Angular CLI: 1.7.3
Node: 6.11.5
OS: win32 x64
Angular: 5.2.9
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.3
@angular-devkit/build-optimizer: 0.3.2
@angular-devkit/core: 0.3.2
@angular-devkit/schematics: 0.3.2
@ngtools/json-schema: 1.2.0
@ngtools/webpack: 1.10.2
@schematics/angular: 0.3.2
@schematics/package-update: 0.3.2
typescript: 2.4.2
webpack: 3.11.0
```

Hi,
I have a problem with an app that I can't really pin down. The structure of our app is as follows:
```
src
|__app
   |__FeatureModules
   |__CommonModules
   |__index.html
   |__main.ts
   |__tsconfig.app.json
   ...
.angular-cli.json
tsconfig.json
...
```
tsconfig.app.json is a default one - it excludes all *.spec.ts files from being built. It does work and they are not being built (I have typos in some just to be sure).

Whenever I run `ng build` it outputs a lot of warnings about *.spec.ts not being built (which is weird, because I would think that it is a desired behavior). It then creates chunks for all those files, with the contents being those warnings. All of the components were created using `ng g c` and are migrated from non-cli Angular components.

Is this working as intended? The "Hello World" app doesn't do that. Why is cli trying to include those files in dist, against tsconfig? Is it somehow related to non-default file structure?

**Comment by robertbaker** [comment]
This was addressed in the latest beta, which changes the project structure to have spec files side by side with the component.

**Comment by robertbaker** [comment]
Also please update your node, it's very old and not supported anymore. Latest release runs faaast.

**Comment by m-sawicki** [comment]
"This was addressed" - meaning it's a bug and it's fixed by beta? Or is it a desired functionality?

Yeah, my node was clearly old, but I don't think it affects my problem in any way. Regardless, I updated it to no apparent changes.

The biggest problem is that those chunks (I suspect) negatively impact my regular chunks creation - feature modules become somehow interconnected and every rebuild takes a long time.

**Comment by m-sawicki** [comment]
@robertbaker Thanks for the help. I tried beta version, but project structure is completely different and the documentation is scarce, so I had to give up on that idea.
In the end I was able to narrow this problem, so for anyone else having the same issue - it was related to SystemJS. I am migrating my app from older setup to CLI and I used SystemJS to load some additional files during runtime. Even though they are very specific, simply using System.import(...) leads to unexpected behavior and loads of chunks being generated (including those for .spec.ts files). I've had to change my approach and eliminate SystemJS all together to get the desired modules.
